prime-angular-frontend/src/app/modules/adjudication/shared/components/review-status-content/review-status-content.component.ts

```diff
@@ -1,16 +1,154 @@
 import { Component, OnInit, Input } from '@angular/core';
 
-import { Reason } from '../../pipes/status-reasons.pipe';
+import { Enrolment } from '@shared/models/enrolment.model';
+import { EnrolmentStatusReason } from '@shared/models/enrolment-status-reason.model';
+import { EnrolmentStatus } from '@shared/models/enrolment-status.model';
+
+class Status {
+  public date: string;
+  public name: string;
+  public code: number;
+  public note: string;
+  public adjudicator: string;
+  public reasons: Reason[];
+}
+
+class Reason {
+  public name: string;
+  public note: string;
+  public isSelfDeclaration: boolean;
+  public question: string;
+}
 
 @Component({
   selector: 'app-review-status-content',
   templateUrl: './review-status-content.component.html',
   styleUrls: ['./review-status-content.component.scss']
 })
 export class ReviewStatusContentComponent implements OnInit {
-  @Input() public reasons: Reason[];
+  // @Input() public enrollee: Enrolment;
+  private _enrollee: Enrolment;
+  public previousStatuses: Status[];
+  public reasons: Reason[];
+
+  // TODO: Currenty we just store this in this place and in the self declaration form
+  // Later on this would be best to be brought out and stored better!
+  private registrationQ = 'Are you, or have you ever been, the subject of an order or a conviction under'
+    + ' legislation in any jurisdiction for a matter that involved improper access to, collection,'
+    + ' use, or disclosure of personal information?';
+  private convictionQ = 'Are you, or have you ever been, subject to any limits, conditions'
+    + ' or prohibitions imposed as a result of disciplinary actions taken by a governing body'
+    + ' of a health profession in any jurisdiction, that involved improper access to, collection,'
+    + ' use, or disclosure of personal information?';
+  private pharmanetQ = 'Have you ever had your access to an electronic health record system,'
+    + ' electronic medical record system, pharmacy or laboratory record system,'
+    + ' or any similar health information system, in any jurisdiction, suspended or cancelled?';
+  private disciplinaryQ = 'Have you ever been disciplined or fired by an employer, or had a contract for your services terminated,'
+    + ' for a matter that involved improper access to, collection, use, or disclosure of personal information?';
+
+
+  @Input() set enrollee(value: Enrolment) {
```

Comment:
Another option instead of using a method with an input decorator is to use the OnChanges hook ngOnChanges then you only need the one input and setter or getter, and ngOnChanges can manage updates/operations for any binding updates.

Reply:
I did this as the actual first pass but ran into problems, and decided to go this route as it worked. Put it on my note pad as on another pass to update.